### PR TITLE
Es6 build

### PR DESCRIPTION
**Pull request by jasonaden**
**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: modules/@angular/animation/src/engine/web_animations/web_animations_player.ts [hunk 1]
```diff
@@ -76,9 +76,9 @@ export class WebAnimationsPlayer implements AnimationPlayer {
       });
 
       if (missingStyleProps.length) {
-        for (let i = 1; i < keyframes.length; i++) {
+        for (var i = 1; i < keyframes.length; i++) {
```
Comment (mhevery) [inline modules/@angular/animation/src/engine/web_animations/web_animations_player.ts]: please add comment as to why

File: modules/@angular/common/common.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export * from './common/index'
```
Comment (mhevery) [inline modules/@angular/common/common.js]: are these files still needed?

File: modules/@angular/compiler/index.ts [hunk 3]
```diff
@@ -9,58 +9,8 @@
...
-export * from './src/output/ts_emitter';
-export * from './src/parse_util';
-export * from './src/schema/dom_element_schema_registry';
-export * from './src/selector';
-export * from './src/style_compiler';
-export * from './src/template_parser/template_parser';
-export {ViewCompiler} from './src/view_compiler/view_compiler';
-export {AnimationParser} from './src/animation/animation_parser';
-export {isSyntaxError, syntaxError} from './src/util';
+export * from './src/compiler';
```
Comment (mhevery) [inline modules/@angular/compiler/index.ts]: why did we have to change this file?

File: modules/@angular/compiler/tsconfig-testing.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
     // Test that we rely on decorator downleveling
     "emitDecoratorMetadata": false,
     "paths": {
-      "@angular/core": ["../../../dist/packages-dist/core"],
+      "@angular/core": ["../../../dist/packages-dist/core/src/core"],
```
Comment (mhevery) [inline modules/@angular/compiler/tsconfig-testing.json]: we should revert these

File: modules/@angular/forms/tsconfig-build.json [hunk 5]
```diff
@@ -8,17 +8,17 @@
     "moduleResolution": "node",
     "outDir": "../../../dist/packages-dist/forms",
     "paths": {
-      "@angular/core": ["../../../dist/packages-dist/core"],
-      "@angular/core/testing": ["../../../dist/packages-dist/core/testing"],
-      "@angular/common": ["../../../dist/packages-dist/common"],
-      "@angular/common/testing": ["../../../dist/packages-dist/common/testing"],
-      "@angular/compiler": ["../../../dist/packages-dist/compiler"],
-      "@angular/compiler/testing": ["../../../dist/packages-dist/compiler/testing"]
+      "@angular/core": ["../../../dist/packages-dist/core/src/core"],
```
Comment (mhevery) [inline modules/@angular/forms/tsconfig-build.json]: revert?

File: tools/@angular/tsc-wrapped/src/main.ts [hunk 6]
```diff
@@ -29,7 +29,7 @@ export type CodegenExtension =
 
 export function main(
     project: string | VinylFile, cliOptions: CliOptions, codegen?: CodegenExtension,
-    options?: ts.CompilerOptions): Promise<any> {
+    options?: ts.CompilerOptions, skipImportRename?: boolean): Promise<any> {
```
Comment (alexeagle) [inline tools/@angular/tsc-wrapped/src/main.ts]: can you please revert this file? We should not need to expand the API surface of tsc_wrapped right? (every change here makes it less likely to upstream into typescript proper)

File: modules/@angular/upgrade/tsconfig-static.json [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    "outDir": "../../../dist/packages-dist/upgrade/static",
+    "target": "es5",
+    "paths": {
+      "@angular/core": ["../../../dist/packages-dist/core"],
+      "@angular/common": ["../../../dist/packages-dist/common"],
+      "@angular/platform-browser": ["../../../dist/packages-dist/platform-browser"],
+      "@angular/platform-browser-dynamic": ["../../../dist/packages-dist/platform-browser-dynamic"]
+    }
+  },
+  "files": [
```
Comment (alexeagle) [inline modules/@angular/upgrade/tsconfig-static.json]: should it be in sync with http://google3/third_party/javascript/angular2/BUILD?l=718 ?

File: integration/README.md [hunk 8]
```diff
@@ -44,7 +43,7 @@ See the `package.json` of the test(s) you're debugging, to see which dist/ folde
 Then run the right `tsc --watch` command to keep those dist folders up-to-date, for example:
 
 ```
-$ ./node_modules/.bin/tsc -p modules/@angular/core/tsconfig-build.json --target es2015 --outDir dist/packages-dist-es2015/core --watch
+$ ./node_modules/.bin/tsc -p modules/@angular/core/tsconfig-build.json --outDir dist/packages-dist/core --watch
```
Comment (alexeagle) [inline integration/README.md]: `--outDir` argument probably not needed anymore

**Review by alexeagle — COMMENTED** [review]
also missing a cleanup in https://github.com/angular/angular/blob/master/scripts/publish/publish-build-artifacts.sh#L141

**Comment by IgorMinar** [comment]
we landed this PR to unblock @matsko - @jasonaden will resolve all the comments in a followup PR.

**Review by IgorMinar — COMMENTED** [review]
approving this but I expect a followup pr to fix all the comments.

File: build.sh [hunk 9]
```diff
@@ -41,20 +43,53 @@ for ARG in "$@"; do
...
+    [[ "$e" == "$1" ]] && return 0;
+  done
+  return 1
+}
+
+NON_MODULE=(
+  platform-browser
+)
+
+moveTypings() {
```
Comment (jasonaden) [inline build.sh]: Done

File: build.sh [hunk 10]
```diff
@@ -111,16 +146,30 @@ fi
...
-  UMD_TESTING_ES5_PATH=${DESTDIR}/bundles/${PACKAGE}-testing.umd.js
-  UMD_STATIC_ES5_PATH=${DESTDIR}/bundles/${PACKAGE}-static.umd.js
-  UMD_UPGRADE_ES5_PATH=${DESTDIR}/bundles/${PACKAGE}-upgrade.umd.js
-  UMD_ES5_MIN_PATH=${DESTDIR}/bundles/${PACKAGE}.umd.min.js
-  UMD_STATIC_ES5_MIN_PATH=${DESTDIR}/bundles/${PACKAGE}-static.umd.min.js
-  UMD_UPGRADE_ES5_MIN_PATH=${DESTDIR}/bundles/${PACKAGE}-upgrade.umd.min.js
+  DEST_MODULE=${DESTDIR}/@angular
+  DEST_BUNDLES=${DESTDIR}/bundles
+
+  # ESM/ES6
```
Comment (jasonaden) [inline build.sh]: Done

File: build.sh [hunk 11]
```diff
@@ -131,34 +180,51 @@ do
...
-  $TSC -p ${SRCDIR}/tsconfig-build.json
-  # ES2015 distro is not ready yet; don't slow down all builds for it
-  # TODO(alexeagle,igorminar): figure out ES2015 story and enable
-  if [[ -n "${EXPERIMENTAL_ES2015_DISTRO}" ]]; then
-    $TSC -p ${SRCDIR}/tsconfig-build.json --target es2015 --outDir ${ES2015_DESTDIR}
-    cp ${SRCDIR}/package.json ${ES2015_DESTDIR}/
-    cp ${PWD}/modules/@angular/README.md ${ES2015_DESTDIR}/
+  echo "======      [${PACKAGE}]: COMPILING: ${TSC} --skipImportRename -p ${SRCDIR}/tsconfig-build.json"
+  if [[ -e ${SRCDIR}/.babelrc || ${PACKAGE} == "compiler" ]]; then
+    $TSC --skipImportRename -p ${SRCDIR}/tsconfig-build.json -outDir ${DEST_MODULE}
```
Comment (jasonaden) [inline build.sh]: Done.

File: build.sh [hunk 12]
```diff
@@ -131,34 +180,51 @@ do
...
-  echo "======      COMPILING: ${TSC} -p ${SRCDIR}/tsconfig-build.json        ====="
-  $TSC -p ${SRCDIR}/tsconfig-build.json
-  # ES2015 distro is not ready yet; don't slow down all builds for it
-  # TODO(alexeagle,igorminar): figure out ES2015 story and enable
-  if [[ -n "${EXPERIMENTAL_ES2015_DISTRO}" ]]; then
-    $TSC -p ${SRCDIR}/tsconfig-build.json --target es2015 --outDir ${ES2015_DESTDIR}
-    cp ${SRCDIR}/package.json ${ES2015_DESTDIR}/
-    cp ${PWD}/modules/@angular/README.md ${ES2015_DESTDIR}/
+  echo "======      [${PACKAGE}]: COMPILING: ${TSC} --skipImportRename -p ${SRCDIR}/tsconfig-build.json"
+  if [[ -e ${SRCDIR}/.babelrc || ${PACKAGE} == "compiler" ]]; then
```
Comment (jasonaden) [inline build.sh]: Done

File: build.sh [hunk 13]
```diff
@@ -131,34 +180,51 @@ do
...
   fi
 
-  if [[ -e ${SRCDIR}/tsconfig-upgrade.json ]]; then
-    echo "======      COMPILING: ${TSC} -p ${SRCDIR}/tsconfig-upgrade.json        ====="
-    $TSC -p ${SRCDIR}/tsconfig-upgrade.json
+  echo "======        Move ${PACKAGE} typings"
+  if [[ -e ${SRCDIR}/.babelrc || -d ${DEST_MODULE} ]]; then
+    rsync -a --exclude=*.js --exclude=*.js.map ${DEST_MODULE}/ ${DESTDIR}/typings
+    mv ${DESTDIR}/typings/index.d.ts ${DESTDIR}/typings/${PACKAGE}.d.ts
+    mv ${DESTDIR}/typings/index.metadata.json ${DESTDIR}/typings/${PACKAGE}.metadata.json
```
Comment (jasonaden) [inline build.sh]: Done

File: modules/@angular/.babelrc [hunk 14]
```diff
@@ -0,0 +1,5 @@
+{
+  "presets": [
+    ["es2015", { "modules": false }]
+  ]
+}
```
Comment (jasonaden) [inline modules/@angular/.babelrc]: Done

File: modules/@angular/common/tsconfig-testing.json [hunk 15]
```diff
@@ -2,11 +2,11 @@
   "extends": "./tsconfig-build",
 
   "compilerOptions": {
+    "outDir": "../../../dist/packages-dist/common",
```
Comment (jasonaden) [inline modules/@angular/common/tsconfig-testing.json]: Done

File: modules/@angular/http/.babelrc-testing [hunk 16]
```diff
@@ -0,0 +1,19 @@
+
+{
+  "presets": ["es2015"],
+  "plugins": [["transform-es2015-modules-umd", {
+    "globals": {
+      "@angular/core": "ng.core",
+      "@angular/compiler": "ng.compiler",
```
Comment (jasonaden) [inline modules/@angular/http/.babelrc-testing]: Done

File: modules/@angular/http/.babelrc [hunk 17]
```diff
@@ -0,0 +1,16 @@
+
+{
+  "presets": ["es2015"],
+  "plugins": [["transform-es2015-modules-umd", {
+    "globals": {
+      "@angular/core": "ng.core",
+      "@angular/compiler": "ng.compiler",
```
Comment (jasonaden) [inline modules/@angular/http/.babelrc]: Done

File: modules/@angular/http/tsconfig-testing.json [hunk 18]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "./tsconfig-build",
   "compilerOptions": {
+    "outDir": "../../../dist/packages-dist/http",
```
Comment (jasonaden) [inline modules/@angular/http/tsconfig-testing.json]: Done

File: modules/@angular/platform-browser-dynamic/tsconfig-testing.json [hunk 19]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "./tsconfig-build",
   "compilerOptions": {
+    "outDir": "../../../dist/packages-dist/platform-browser-dynamic",
```
Comment (jasonaden) [inline modules/@angular/platform-browser-dynamic/tsconfig-testing.json]: Done

File: modules/@angular/platform-browser/tsconfig-testing.json [hunk 20]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "./tsconfig-build",
   "compilerOptions": {
+    "outDir": "../../../dist/packages-dist/platform-browser",
```
Comment (jasonaden) [inline modules/@angular/platform-browser/tsconfig-testing.json]: Done

File: modules/@angular/upgrade/.babelrc-static [hunk 21]
```diff
@@ -0,0 +1,12 @@
+
```
Comment (jasonaden) [inline modules/@angular/upgrade/.babelrc-static]: Done

File: tools/@angular/tsc-wrapped/src/main.ts (hunk 6 above)
Comment (jasonaden) [inline tools/@angular/tsc-wrapped/src/main.ts]: Done

File: .babelrc [hunk 22]
```diff
@@ -0,0 +1,4 @@
+{
+  "plugins": ["transform-es2015-modules-umd"],
+  "presets": ["es2015"]
+}
```
Comment (jasonaden) [inline .babelrc]: Done

File: integration/README.md (hunk 8 above)
Comment (jasonaden) [inline integration/README.md]: Correct. Removed
